### PR TITLE
Made path's relative to handle project existing anywhere

### PR DESCRIPTION
**Pull request by tyschacht**
The existing project paths appear to assume the project exists at the root of the drive in a directory named 'ApexSharp'.   This adjusts the paths to handle relative paths that work anywhere the project is located.

If i'm missing something obvious that causes issues across systems/platforms, please ignore.  :)

### CODE REVIEW

**Comment by tyschacht** [comment]
This is a combo of the last pull request, with a fix (removed custom classes from sObject request list) and an addition to the sObject list (recordtype).

If the admin has customized their objects (as I did), they could have a reference of 'RecordType' on one of the included salesforce objects which will cause compilation errors (and confusion) after the first run.  I went ahead and added 'RecordType' to the default list on the first sync.

**Comment by tyschacht** [comment]
Fixed a simple merge conflict.

**Comment by yallie** [comment]
Hi @tyschacht,

I like the idea to get rid of the hardcoded paths! In fact, one of the previous versions of the ApexSharp demo used paths relative to the current assembly. I don't know exactly why @jayonsoftware  reverted to the hardcoded paths, perhaps it was related to Linux?

Hovewer, there is an issue with the ".." notation. It refers to the current directory, but the current directory can be easily changed anytime. If some code changes the current directory after the `Setup.Init()` call, I'm afraid things may break.

@jayonsoftware, what you think?

File: Demo/Setup.cs [hunk 1]
```diff
@@ -58,12 +58,12 @@ public static bool Init()
...
                     ConnectionUtil.Session = new ApexSharp().
-                         SalesForceUrl("https://login.salesforce.com/")
+                         SalesForceUrl("https://test.salesforce.com/")
                         .AndSalesForceApiVersion(40)
                         
-                        .WithUserId("SF Login ID")
-                        .AndPassword("Password")
-                        .AndToken("Token")
+                        .WithUserId("allison@healthclx.com.test")
+                        .AndPassword("healthclxer007")
```
Comment (yallie) [inline Demo/Setup.cs]: I don't think that was intentional?
You hardly want to put your password in the source code repository :)

**Comment by jayonsoftware** [comment]
The goal was to use JSON as the init file. Let me make that default

**Comment by jayonsoftware** [comment]
Also Program.cs or any files in the Demo project no need to update git as its an example

File: Demo/Setup.cs (hunk 1 above)
Comment (tyschacht) [reply Demo/Setup.cs]: definitely not.  thanks

**Comment by tyschacht** [comment]
@jayonsoftware  -- As a new user coming in (from your appearance on .Net Rocks, BTW) the fact that the Demo project did not work "out-of-the-box" was a slight barrier.    That's the main reason I submitted those changes as they seemed to be things that any new user would have to do just to get the demo working.  

Is it cool if I do an update to the README.md for the Demo to clarify any sticking points I ran into during my introduction to the project?   I think that would greatly help new users coming on board.

**Comment by jayonsoftware** [comment]
I agree I am refactoring the Demo project. Feel free to update the readme.md, but can you work on the dev branch as that is the latest.

**Comment by jayonsoftware** [comment]
I have updated the dev branch with relative paths. Going to test on linux and mac

**Comment by jayonsoftware** [comment]
@tyschacht I have updated the readme.md, feel free to update it more.

**Pull request merged**